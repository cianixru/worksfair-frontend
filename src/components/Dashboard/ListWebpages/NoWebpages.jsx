import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NoWebpages = ({ user }) => (
  <li>
    <div className="message is-warning">
      <div className="message-body">
        <p className="is-size-5">
          You have no active webpages
        </p>
        { user.confirmed_account &&
          <Link
            to={`/dashboard/${user.username}/webpages/new/basic-info`}
            className="button is-link is-outlined  margin-top-25">
            Create a Webpage
          </Link>
        }
      </div>
    </div>
  </li>
);

NoWebpages.propTypes = {
  user: PropTypes.object,
};

export default NoWebpages;
