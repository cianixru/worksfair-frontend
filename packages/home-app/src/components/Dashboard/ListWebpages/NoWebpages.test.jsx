import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { BrowserRouter as Router } from 'react-router-dom';

import NoWebpages from './NoWebpages';

afterEach(cleanup);

describe('NoWebpages.jsx', () => {
  test('Should display the NoWebpages Component', () => {
    const component = render(
      <Router>
        <NoWebpages user={{
          username: "TheoOkafor",
          confirmed_account: false
        }} />
      </Router>
    );
    expect(component).toBeDefined();
    expect(component).toMatchSnapshot();
  });
});
