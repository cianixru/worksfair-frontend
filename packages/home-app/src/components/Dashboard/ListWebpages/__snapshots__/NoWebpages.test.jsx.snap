// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoWebpages.jsx Should display the NoWebpages Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <div
          class="message is-warning"
        >
          <div
            class="message-body"
          >
            <p
              class="is-size-5"
            >
              You have no active webpages
            </p>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <div
        class="message is-warning"
      >
        <div
          class="message-body"
        >
          <p
            class="is-size-5"
          >
            You have no active webpages
          </p>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
